### PR TITLE
LibWasm: Remove `Module::functions`

### PR DESCRIPTION
**Pull request by dzfrias**
`Module::functions` created clones of all of the functions in the module. It provided a _slightly_ better API, but ended up costing around 40ms when instantiating spidermonkey.

This brings us from 180ms -> 140ms when instantiating spidermonkey.

### CODE REVIEW

File: Userland/Libraries/LibWasm/Printer/Printer.cpp [hunk 1]
```diff
@@ -541,33 +541,6 @@ void Printer::print(Wasm::Module const& module)
     print(")\n");
 }
 
-void Printer::print(Wasm::Module::Function const& func)
```
Comment (alimpfard) [inline Userland/Libraries/LibWasm/Printer/Printer.cpp:544]: _something_ should print the functions' contents.

**Review by alimpfard — COMMENTED** [review]
Generally, this is fine, but please keep the printer's behaviour.

File: Userland/Libraries/LibWasm/Printer/Printer.cpp (hunk 1 above)
Comment (dzfrias) [reply Userland/Libraries/LibWasm/Printer/Printer.cpp:544]: I removed `Function` as a whole

Comment (alimpfard) [reply Userland/Libraries/LibWasm/Printer/Printer.cpp:544]: Ah, that's true, it's being printed through `code.func()`.

**Review by alimpfard — APPROVED** [review]

**Pull request merged**